cpp/arcticdb/version/symbol_list.cpp

```diff
@@ -216,27 +241,31 @@ static const StreamId compaction_id {CompactionId};
         return output;
     }
 
-    void SymbolList::delete_keys(
-            std::shared_ptr<Store> store, const KeyVector& lists) {
-        std::vector<VariantKey> vars;
-        vars.reserve(lists.size());
-        for(const auto& atom_key: lists)
-            vars.emplace_back(VariantKey{atom_key});
-        store->remove_keys(vars).get();
+    folly::Future<std::vector<Store::RemoveKeyResultType>>
+    SymbolList::delete_keys(const std::shared_ptr<Store>& store, KeyVector&& to_remove, const AtomKey& exclude) {
+        std::vector<VariantKey> variant_keys;
+        variant_keys.reserve(to_remove.size());
```

Comment:
Are we making a complete copy of all the keys except one to remove one of them?

Reply:
This is the Atom->Variant Key conversion and filtering folded into one. I've done my best with `std::move` below.

Reply:
I think it would be better to make the types consistent. VariantKey is a later addition, but generally speaking we are converging on using it everywhere to avoid this sort of copying conversion. But I guess we can leave it for a future change